TestFoundation/TestFileManager.swift

```diff
@@ -202,6 +206,62 @@ class TestFileManager : XCTestCase {
         ignoreError { try fm.removeItem(atPath: tmpDir.path) }
     }
 
+    func test_isReadableFile() {
+        let fm = FileManager.default
+        let path = NSTemporaryDirectory() + "test_fileAttributes\(NSUUID().uuidString)"
+
+        XCTAssertTrue(fm.createFile(atPath: path, contents: Data(), attributes: nil))
+
+        do {
+            let attrs = try fm.attributesOfItem(atPath: path)
+            let permissions = attrs[FileAttributeKey.posixPermissions] as! UInt16
+            let fileIsReadableFile = (permissions & S_IRUSR == S_IRUSR)
+            let fmIsReadableFile = fm.isReadableFile(atPath: path)
+            XCTAssertTrue(fileIsReadableFile == fmIsReadableFile)
+        } catch let e {
+            XCTFail("\(e)")
+        }
+    }
+
+    func test_isWritableFile() {
+        let fm = FileManager.default
+        let path = NSTemporaryDirectory() + "test_fileAttributes\(NSUUID().uuidString)"
+
+        XCTAssertTrue(fm.createFile(atPath: path, contents: Data(), attributes: nil))
+
+        do {
+            let attrs = try fm.attributesOfItem(atPath: path)
+            let permissions = attrs[FileAttributeKey.posixPermissions] as! UInt16
+            let fileIsReadableFile = (permissions & S_IWUSR == S_IWUSR)
+            let fmIsReadableFile = fm.isReadableFile(atPath: path)
+            XCTAssertTrue(fileIsReadableFile == fmIsReadableFile)
+        } catch let e {
+            XCTFail("\(e)")
+        }
+    }
+
+    func test_isExecutableFile() {
+        let fm = FileManager.default
+        let path = NSTemporaryDirectory() + "test_fileAttributes\(NSUUID().uuidString)"
+
+        XCTAssertTrue(fm.createFile(atPath: path, contents: Data(), attributes: nil))
+
+        do {
+            let attrs = try fm.attributesOfItem(atPath: path)
+            let permissions = attrs[FileAttributeKey.posixPermissions] as! UInt16
+            let fileIsReadableFile = (permissions & S_IXUSR == S_IXUSR)
+            let fmIsReadableFile = fm.isReadableFile(atPath: path)
+            XCTAssertTrue(fileIsReadableFile == fmIsReadableFile)
+        } catch let e {
+            XCTFail("\(e)")
+        }
+    }
+
+    func test_isDeletableFile() {
+        // TODO: Implement test
+        // how to test?
+    }
```

Comment:
You could create a file and check its is deletable. Also check a file that should fail eg `/dev/null` and also check `/` to validate the parent checking code

Reply:
Just to confirm, we would expect the created file to be deletable, correct?

Reply:
Yes, thats correct. You should be able to create a file that fails by putting it into a subdirectory with no permissions:

eg 
```
mkdir dir1
touch dir1/file1
chmod 000 dir1

isDeletableFile("dir1/file1") -> false

chmod 755 dir1

isDeletableFile("dir1/file1") -> true
```